packages/react-components/src/components/Picker/TriggerBody.tsx

```diff
@@ -30,10 +31,22 @@ export const TriggerBody: React.FC<ITriggerBodyProps> = ({
   items,
   type,
   iconSize,
+  clearSearchAfterSelection,
   onItemRemove,
   onFilter,
 }) => {
   const shouldDisplaySearch = isOpen && !isSearchDisabled;
+  const inputRef = React.useRef<HTMLInputElement>(null);
+
+  React.useEffect(() => {
+    if (clearSearchAfterSelection) {
+      onFilter('');
+
+      if (inputRef.current) {
+        inputRef.current.value = '';
+      }
+    }
+  }, [items, clearSearchAfterSelection]);
```

Comment:
Is it intended that `items` are in the dependency array and `onFilter` is not?

Reply:
Yes. `items` will change after selection so it should triggers `useEffect`. If we put `onFliter` it will break search functionality as this handler is triggered when search input value changes.